src/librbd/mirror/GroupGetInfoRequest.cc

```diff
@@ -172,7 +187,7 @@ void GroupGetInfoRequest<I>::handle_get_last_mirror_snapshot_state(int r) {
       // XXXMG: check primary_mirror_uuid matches?
       switch (ns->state) {
       case cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY:
-        if (it->state == cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE) {
+        if (!is_group_snapshot_complete(it->state, ns->complete)) {
```

Comment:
Because of `if (complete == cls::rbd::MIRROR_GROUP_SNAPSHOT_COMPLETE_IF_CREATED && ...` check in the `is_group_snapshot_complete()` isn't this mirror group snapshot check here looks problematic ?